### PR TITLE
check if MyParcel method is selected

### PR DESCRIPTION
**Pull request by RichardPerdaan**
this solution is for users with MagePlaza OneStepCheckout version 2.6.1

### CODE REVIEW

File: Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php [hunk 1]
```diff
@@ -67,7 +81,7 @@ public function execute(\Magento\Framework\Event\Observer $observer)
             $order->setData(self::FIELD_TRACK_STATUS, __('⚠️&#160; Please check address'));
         }
 
-        if ($quote->hasData(self::FIELD_DELIVERY_OPTIONS)) {
+        if ($quote->hasData(self::FIELD_DELIVERY_OPTIONS) && $this->isMyParcelMethod( $quote )) {
```
Comment (reindert-vetter) [inline Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php]:
```suggestion
        if ($quote->hasData(self::FIELD_DELIVERY_OPTIONS) && $this->isMyParcelMethod($quote)) {
```

File: Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php [hunk 2]
```diff
@@ -78,4 +92,45 @@ public function execute(\Magento\Framework\Event\Observer $observer)
         return $this;
     }
 
+    /**
+     * @param $quote
+     *
+     * @return bool
+     */
+    private function isMyParcelMethod( $quote ) {
```
Comment (reindert-vetter) [inline Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php]:
```suggestion
    private function isMyParcelMethod($quote) {
```

File: Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php [hunk 3]
```diff
@@ -78,4 +92,45 @@ public function execute(\Magento\Framework\Event\Observer $observer)
...
+        $myParcelMethods = array_keys( Carrier::getMethods() );
```
Comment (reindert-vetter) [inline Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php]:
```suggestion
        $myParcelMethods = array_keys(Carrier::getMethods());
```

File: Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php [hunk 4]
```diff
@@ -78,4 +92,45 @@ public function execute(\Magento\Framework\Event\Observer $observer)
...
+        $shippingMethod  = $quote->getShippingAddress()->getShippingMethod();
+        /**
+         * @todo explode and use second part
+         * $shippingMethod = ...
+         */
+
+        if ( $this->array_like( $shippingMethod, $myParcelMethods ) ) {
```
Comment (reindert-vetter) [inline Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php]:
```suggestion
        if ($this->array_like($shippingMethod, $myParcelMethods)) {
```

File: Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php [hunk 5]
```diff
@@ -78,4 +92,45 @@ public function execute(\Magento\Framework\Event\Observer $observer)
...
+            return true;
+        }
+
+        if ( $this->array_like( $shippingMethod, $this->parentMethods ) ) {
```
Comment (reindert-vetter) [inline Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php]:
```suggestion
        if ($this->array_like($shippingMethod, $this->parentMethods)) {
```

File: Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php [hunk 6]
```diff
@@ -78,4 +92,45 @@ public function execute(\Magento\Framework\Event\Observer $observer)
...
+     */
+    private function array_like($input, $data) {
+        $result = array_filter($data, function ($item) use ($input) {
+            if (stripos($input, $item) !== false) {
+                return true;
+            }
+            return false;
+        });
+
+        return count( $result ) > 0;
```
Comment (reindert-vetter) [inline Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php]:
```suggestion
        return count($result) > 0;
```

**Comment by RichardPerdaan** [comment]
Hi Kroellie,

With MagePlaza OneStepCheckout version 2.6.1 we have noticed that the PostNL locations do not get through to the Magento 2 admin.
The following message you can see inside the order: 

000000045 _ orders _ operations _ sales _ magento admin 2018-10-19 11-34-24

You will not have this problem with older versions of the MagePlaza OneStepCheckout.

File: Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php (hunk 1 above)
Comment (RichardPerdaan) [reply Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php]: Done!

File: Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php (hunk 3 above)
Comment (RichardPerdaan) [reply Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php]: Done!

File: Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php (hunk 5 above)
Comment (RichardPerdaan) [reply Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php]: Done!

File: Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php (hunk 6 above)
Comment (RichardPerdaan) [reply Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php]: Done!

File: Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php (hunk 1 above)
Comment (RichardPerdaan) [reply Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php]: Done!

File: Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php [hunk 7]
```diff
@@ -78,4 +92,45 @@ public function execute(\Magento\Framework\Event\Observer $observer)
...
+     *
+     * @return bool
+     */
+    private function isMyParcelMethod($quote) {
+        $myParcelMethods = array_keys(Carrier::getMethods());
+        $shippingMethod  = $quote->getShippingAddress()->getShippingMethod();
+        /**
+         * @todo explode and use second part
+         * $shippingMethod = ...
+         */
```
Comment (reindert-vetter) [inline Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php]: Dit weghalen, halve als dit toch nodig is. Even testen of dit werkt

Comment (RichardPerdaan) [reply Model/Quote/SaveOrderBeforeSalesModelQuoteObserver.php]: Done!

**Pull request merged**